src/cljs/vanilla/add_widget.cljs

```diff
@@ -173,43 +174,86 @@
 
 
 
-(defn add-widget-modal [is-active]
+(defn add-widget-modal
+  ""
+  [is-active]
   (let [services (rf/subscribe [:services])
         selected (rf/subscribe [:selected-service])
         chosen-widget (rf/subscribe [:selected-widget])
         widget-cards (rf/subscribe [:all-widget-types])
         compatible-selection (rf/subscribe [:compatible-selections])]
     (fn []
-      [:div.modal (if @is-active {:class "is-active"})
-       [:div.modal-background]
-       [:div.modal-card
-        [:header.modal-card-head
-         [:p.modal-card-title "Add Data Source"]
-         [:button.delete {:aria-label "close"
-                          :on-click   #(reset! is-active false)}]]
 
-        ;[:section.model-card-body
-        ; [:p (str "selected " @selected)]
-        ; [:p (str "widget " @chosen-widget)]]
-
-        [:section.modal-card-body
-         [service-list @services @selected]]
-
-        [:section.modal-card-body
-         [widget-list @widget-cards @selected @chosen-widget]]
-
-        [:footer.modal-card-foot
-         [:button.button.is-success {:disabled (not @compatible-selection)
-                                     :on-click #(do
-                                                  ;(prn "picked widget " @chosen-widget @selected)
-                                                  (add-widget (:name @chosen-widget) @selected)
-                                                  (reset! is-active false))} "Add"]
-         [:button.button {:on-click #(reset! is-active false)} "Cancel"]]]])))
+      (modal/create-modal {:is-active             is-active
+                           :title                 "Add Data Source"
+                           :modal-body-list       (vector [service-list @services @selected]
```

Comment:
don't need to "call" (vector). you can just [ [thing-1] [thing-2] ] as a literal

Reply:
Yeah, I had it list and vector to test around with the looping of the hiccup elements. I'll change it to normal idiomatic vector declaration now.